accepted/0000-yomi-initial-integration.md

```diff
@@ -0,0 +1,361 @@
+
+The current `yomi-formula` is using this mechanism to split the pillar
+data into six sections:
+
+- yomi: general configuration data, like use `kexec` instead of
+  `reboot` after the installation, or some `systemd-firstboot` data.
+
+- yomi-bootloader: indicate the device where the bootloader will be
+  installed, and the parameters send to the kernel.
+
+- yomi-services: list of services that will be enabled, and is
+  `salt-minion` data will be copied into the new system.
+
+- yomi-software: repository registering and packages installation.
+
+- yomi-storage: device partition, LVM or RAID configuration, and file
+  systems applied.
+
+- yomi-users: creation of initial users, passwords and certificates.
+
+As we will see in future sections, those forms contains some
+overlapping with other salt-forms (the locale configuration that
+`systemd-firstboot` requires), and with other subsystems in Uyuni
+(like the software section).
+
+## Highstate and Workflow
+
+Installing a new system will be equivalent of applying a highstate to
+a system booted from the Yomi image.
+
+The current Yomi image is a JeOS based image with a `salt-minion`,
+some systemd overwrites that will help during the initial
+configuration, and the CLI tools required when a new installation is
+performed. You can find more information here:
+https://github.com/openSUSE/yomi#the-iso-image
+
+The current Yomi states are aware of differentiating when the system
+is booting from an external image (like the Yomi image), from when the
+boot comes from the local device. This is used to decide when the
+installation of the node makes sense or not when a highstate is
+re-applied.
+
+This means that we can set all the pillar data for the node, without
+risk of breaking or damaging already installed nodes when the Yomi
+state is in place for those systems.
+
+This open the question about the workflow that will be in place when a
+new system is allocated. The current prototype was tested like this
+one:
+
+1) Boot the new node via the Yomi ISO image. The local DHCP server
+   assign a FQDN to the system, and this is used automatically as a
+   minion ID.
+
+   (note: the master address was indicated via a Grub command line
+   that an overwrite in the image can read and interpret, how the new
+   minion will automatically discover the master is an open question
+   for me yet)
+
+2) Accept the Salt key in the manager node (Uyuni 4.0.2), and register
+   the system.
+
+3) Assign to the new node all the Yomi states (Installer group),
+   fulfill the SaltForms and apply the highstate to the new node.
+
+Once the node reboot, the highstate can be re-applied without
+consequences, as the state detect early that the root filesystem
+device is already mounted.
+
+Some minor bugs in Uyuni (bsc#1150586, bsc#115089, bsc#115090,
+bsc#1152673, bsc#1152767) makes this process less direct, but for all
+except bsc#1152673 Yomi is providing a workaround. The remaining bug
```

Comment:
For the bsc#1152673 we have a workaround in the [RegisterMinionEventMessageAction.java#L505](https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java#L505)

Reply:
IIUC the highstate will succeed but the SLS rendering will continue to fail?

This somehow invalidate the manual application of the hs from the user, something that can be required later if the grains proposal is approved.

Reply:
As this last bug was confirmed, we can wait for the resolution before editing this part of the proposal.